### ISSUE TITLE
Can't access remult in Next JS API route

### ISSUE THREAD

**Issue by guywuy** [remult/remult#178]
I'm trying to create a simple Next JS API route that queries a remult repo and returns some items (in `/pages/api/overdue.ts`). I tried accessing the remult repo in the API handler directly with 
`remult.repo(Chore);` but got the error - `Error: remult object was requested outside of a valid context, try running it within initApi or a remult request cycle`

I moved the query to a backend method and added it to the controller as per the remult next js tutorial: 
```ts
export default remultNext({
    getUser: async req => getUserById((await getToken({ req }))?.sub),
    entities: [Chore],
    controllers: [ChoresController],
    dataProvider: createPostgresConnection()
})
```

but still got the same error.
I then saw [this issue and reply](https://github.com/remult/remult/issues/145#issuecomment-1330145953) so added in the optional `remult`  parameter to the backend method:

```ts
@BackendMethod({ allowed: Allow.everyone })
    static async getAllOverdue(remult?: Remult) {
        const choreRepo = remult!.repo(Chore)
        const due = await choreRepo.find({
            where: {
                due: { "<=": Date.now() }
            },
            orderBy: {
                due: "asc",
            },
        })
        return due;
    }
 ```
 
 But when calling the backend method from in the API handler, 
```ts
export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>
) {
  const todo = await ChoresController.getAllOverdue();

  res.status(200).json({ chores: todo })
}
```
it is still not working, with the error `TypeError: Cannot read properties of undefined (reading 'repo')` and pointing to the line `const choreRepo = remult!.repo(Chore)` in the backend method so it seems the remult object is not being passed correctly.

---

If I'm missing something let me know, or if there is an easier way to just retrieve the list of items via an API endpoint please let me know. Thanks.

**Comment by noam-honig** [comment]
Hi @guywuy 

Back to basics - you don't need the backend method in this case, for a regular next route.

The problem you encountered was that when being called from a handler, remult was not initialized yet.

to initialize it, wrap your code with a "withRemult" call.
```ts
import api from "./[...remult]";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>
) {
api.withRemult(req, res, async () => {
   const choreRepo = remult.repo(Chore)
   const due = await choreRepo.find({
            where: {
                due: { "<=": Date.now() }
            },
            orderBy: {
                due: "asc",
            },
        })
  res.status(200).json({ chores: due})
});
}
```


This actually gives us an idea for an improved syntax for remult with next in such cases - what do you think about:

```ts
import api from "./[...remult]";

export default api.handler(
  req: NextApiRequest,
  res: NextApiResponse<Data>
) {

   const choreRepo = remult.repo(Chore)
   const due = await choreRepo.find({
            where: {
                due: { "<=": Date.now() }
            },
            orderBy: {
                due: "asc",
            },
        })
  res.status(200).json({ chores: due})
});

```


What do you think?

**Comment by guywuy** [comment]
Wow thanks for the super quick response :)

Yes that works perfectly, I missed that `withRemult` option. Many thanks. Maybe could be documented in the next tutorial

I like the new syntax, simpler 👍

**Comment by noam-honig** [comment]
Hi @guywuy,

Just released version 0.20.0 with the new syntax we've discussed. It is also detailed now in the tutorial:

https://remult.dev/tutorials/react-next/appendix-1-get-server-side-props.html#using-remult-in-a-next-js-api-handler
